### PR TITLE
Add more description for deployment/architecture

### LINKED ISSUE #128: Elaborate on Deployment document
Current document just includes figures for the deployment model and is without explanation.

It should probably also named "Architecture".

### PR DESCRIPTION
**Pull request by Huy-Ngo**
Expectedly resolve #128 

Beware: I write this in my sleep-deprived mind so words probably are not making sense. Correct grammar and writing if needed.

### CODE REVIEW

File: docs/source/meth/arch.rst [hunk 1]
```diff
@@ -1,59 +1,97 @@
...
+
+   node Server as server_3 {
+      component Acanban as acanban_3
+      component Hypercorn as hypercorn_3
    }
 
    node "Database Server" as db_server {
       database "RethinkDB" as db
       component IPFS
    }
```
Comment (McSinyx) [inline docs/source/meth/arch.rst:77]: I think I didn't notice this earlier, when scaled IPFS and RethinkDB are to be run in clusters (and there's very little reason to run them on the same node when we have many machines.

File: docs/source/meth/arch.rst [hunk 2]
```diff
@@ -1,59 +1,97 @@
...
       component NGINX as nginx <<load balancer>>
-      artifact "/etc/nginx/sites-available/acanban" as nginx_conf
    }
 
-   node Server {
-      artifact "acanban.service" as systemd_conf
-      component Acanban
-      component Hypercorn
-      node systemd <<daemon>>
+   node Server as server_1 {
```
Comment (McSinyx) [inline docs/source/meth/arch.rst:59]: To avoid the confusion discussed yesterday on Matrix, let's call these nodes *Web server*s.

Comment (Huy-Ngo) [reply docs/source/meth/arch.rst:59]: Hmm isn't hypercorn the web server?

File: docs/source/meth/arch.rst (hunk 1 above)
Comment (Huy-Ngo) [reply docs/source/meth/arch.rst:77]: So it's a many to many relation plus the database servers are connected with each other?

image

**Comment by McSinyx** [comment]
It's the ASGI server.  I still think we need to elaborate on the *GI
model because e.g. for PHP you don't have the server and the application
as separate components.

**Comment by Huy-Ngo** [comment]
> I still think we need to elaborate on the *GI

I myself am very ignorant of that; it's definitely worth a section/subsection plus a glossary entry.

File: docs/source/meth/arch.rst (hunk 1 above)
Comment (McSinyx) [reply docs/source/meth/arch.rst:77]: Yes, also it's been so long since OOSAD that we forgot about the relation is part of the language UM.

**Comment by McSinyx** [comment]
Imma take it up on the ASGI then and move it out of the scope of this PR.

File: docs/source/meth/arch.rst (hunk 1 above)
Comment (Huy-Ngo) [reply docs/source/meth/arch.rst]: image
I'm not sure if this is good enough.

Comment (McSinyx) [reply docs/source/meth/arch.rst]: I meant the `1..*` thing to simplify the figure.

Comment (Huy-Ngo) [reply docs/source/meth/arch.rst]: Yea, why did I remove that. Will do tomorrow though.

**Review by McSinyx — APPROVED** [review]

**Pull request merged**